### ISSUE TITLE
Bizzarely specific hang

### ISSUE THREAD

**Issue by shusain93** [uranusjr/macdown#433]
Hey, so I found a weird hang bug with even weirder reproduction instructions. Occours with version 0.5 (670) and [here's the crash log] (https://gist.github.com/shusain93/3d8ba0a749153fd9c133)

1. Open a new document and type something in it without saving or anything
2. Hit Command-Q
3. At the "Don't save, Cancel, Save As" prompt, hit the escape key. Hitting the cancel button does not produce this bug
4. Click the close button on the document
5. Application hangs

**Comment by Jmuccigr** [comment]
That doesn't happen to me. Instead when I hit escape or the Cancel button, the eve dialogue closes and the app quits anyway.

The app shouldn't quit in either case.

So that's still a bug, but not the one you report.

MacDown 0.5. OS X 10.9.5.

**Comment by shusain93** [comment]
Maybe it's caused by a newer form of OSX. I have 10.10.5 (14F27)

**Comment by Jmuccigr** [comment]
@shusain93,

Does the app stay open if you hit the Cancel button?

**Comment by shusain93** [comment]
Yup. [Here's a video](http://youtu.be/TcgCwmA7Haw) of what I did. The screen capture didn't capture the beach ball, but as soon as you see the close button get stuck down and the cursor moving everywhere, the app has hung

**Comment by shusain93** [comment]
Still hasn't been fixed, nor labeled. Should be labeled as a bug

**Comment by uranusjr** [comment]
I think something already went wrong when you cancelled the ⌘D dialog, since the MacDown menu kept showing highlighted after that. It shouldn’t. But I couldn’t reproduce that either. I need at least some confirmation this is not specific to you to label this as a bug. I have no ways to fix this if it’s not reproducible anyway…

**Comment by Jmuccigr** [comment]
@uranusjr, can you reproduce the bug I reported of hitting escape canceling the save AND quitting?

I'm on Mavericks with MD 0.5.2.

**Comment by Jmuccigr** [comment]
Anything on this? Still doing it on Mavericks and Version 0.5.5.1 (717) of MacDown.

**Comment by shusain93** [comment]
Nope. Issue still hasn't been tagged as an issue yet, still doing the exact same thing as it was pretty much a year ago.
